Source/DafnyLanguageServer/Language/GhostStateDiagnosticCollector.cs

```diff
@@ -20,18 +23,25 @@ public class GhostStateDiagnosticCollector : IGhostStateDiagnosticCollector {
     private const string GhostStatementMessage = "Ghost statement";
 
     private readonly GhostOptions options;
-
-    public GhostStateDiagnosticCollector(IOptions<GhostOptions> options) {
+    private readonly ILogger<GhostStateDiagnosticCollector> logger;
+    public GhostStateDiagnosticCollector(IOptions<GhostOptions> options, ILogger<GhostStateDiagnosticCollector> logger) {
+      this.logger = logger;
       this.options = options.Value;
     }
 
     public IEnumerable<Diagnostic> GetGhostStateDiagnostics(SymbolTable symbolTable, CancellationToken cancellationToken) {
       if (!options.MarkStatements) {
         return Enumerable.Empty<Diagnostic>();
       }
-      var visitor = new GhostStateSyntaxTreeVisitor(symbolTable.CompilationUnit.Program, cancellationToken);
-      visitor.Visit(symbolTable.CompilationUnit.Program);
-      return visitor.GhostDiagnostics;
+
+      try {
+        var visitor = new GhostStateSyntaxTreeVisitor(symbolTable.CompilationUnit.Program, cancellationToken);
+        visitor.Visit(symbolTable.CompilationUnit.Program);
+        return visitor.GhostDiagnostics;
+      } catch (Exception e) {
+        logger.LogDebug(e, "encountered an exception while getting ghost state diagnostics of {Name}", symbolTable.CompilationUnit.Name);
```

Comment:
This will not be visible to users, right?  Aren't these logs only written to disk in a somewhat hidden location?

Reply:
Yes they are written next to DafnyLanguageServer.dll, and only if an option is activated. 
But this kind of error can happen if the language server recovered from problematic resolving, and leaving the program in a state that the Ghost Diagnostic Collector is not assuming, so that's why it's less important.